.github/workflows/linux.yml

```diff
@@ -16,474 +16,490 @@ concurrency:
+          python -m pytest -vv ./${{ matrix.test.cmd }}
+        working-directory: ${{ env.SRC_DIR }}
+        
+  genai_samples_tests:
+    name: Samples Tests - ${{ matrix.build-type }}
+    strategy:
+      fail-fast: false
+      matrix:
+        build-type: [Release]
+    needs: [ openvino_download, genai_build_samples, genai_build_wheel ] 
+    timeout-minutes: 45
+    defaults:
+      run:
+        shell: bash
+    runs-on: aks-linux-2-cores-8gb
+    container:
+      image: openvinogithubactions.azurecr.io/ov_test/ubuntu_22_04_x64:${{ needs.openvino_download.outputs.docker_tag }}
+      volumes: 
+        - /mount:/mount
+        - ${{ github.workspace }}:${{ github.workspace }}
 
-      - name: Build samples (Release)
-        if: ${{ 'Release' == matrix.build-type }} # build_samples enforces Release build
-        run: |
-          ${OV_INSTALL_DIR}/samples/cpp/build_samples.sh -i ${{ github.workspace }}/s\ pace
+    env:
+      INSTALL_DIR: ${{ github.workspace }}/install
+      SRC_DIR: ${{ github.workspace }}/src
+      BUILD_DIR: ${{ github.workspace }}/build
+      MODELS_DIR: ${{ github.workspace }}/models
+        
+    steps:
+      - name: Clone openvino.genai
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: ${{ env.SRC_DIR }}
+          submodules: recursive
+          
+      - name: Download OpenVINO Package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ${{ needs.openvino_download.outputs.ov_artifact_name }}
+          path: ${{ env.INSTALL_DIR }}
+          merge-multiple: true
 
-      - name: Build samples (Debug)
-        if: ${{ 'Release' != matrix.build-type }}
+      - name: Download Build Artifacts
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          pattern: genai_+(archive_${{ matrix.build-type }}|wheels|samples_${{ matrix.build-type }})
+          path: ${{ env.INSTALL_DIR }}
+          merge-multiple: true
+      
+      - name: Extract Artifacts
         run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${OV_INSTALL_DIR}/samples/cpp/ -B ./samples\ build/ && cmake --build ./samples\ build/ --config ${{ matrix.build-type }} -j
-          cmake --install ./samples\ build/ --config ${{ matrix.build-type }} --component samples_bin --prefix s\ pace
-
-      - name: Test C++ samples (greedy_causal_lm)
+          pigz -dc genai.tar.gz | tar -xf - -C ${{ env.INSTALL_DIR }}
+          pigz -dc genai_samples.tar.gz | tar -xf - -C ${{ env.INSTALL_DIR }}
+        working-directory: ${{ env.INSTALL_DIR }}
+        
+      - name: Install GenAI Wheels
+        run: python -m pip install ${{ env.INSTALL_DIR }}/wheels/openvino_genai*.whl openvino-tokenizers[transformers] ${{ needs.openvino_download.outputs.ov_wheel_source }} --force-reinstall
```

Comment:
What is `--force-reinstall` for? The environment is supposed to be clean

Reply:
That is a common practice, just do not keep in mind if the environment clean or not.  
For example if someone would like to swap steps (requirements and wheels installation)...